### ISSUE TITLE
Add support for image uploads for Lounge and Game Messages

### ISSUE THREAD

**Issue by mikelax** [mikelax/forgingadventures#85]
Overview
===
As a user, I want to be able to insert an image into either a lounge or game message. 

Requirements
===
For each editor, there should be a new icon added to the editor toolbar. It should use the semantic "image" icon. Clicking this should bring up the file browser for the user to select an image to upload. 

The image should be inserted into the editor at the cursor location. 
After the message is created or saved, the image should be displayed inline to the message. 
A user viewing the message can click the image to open a larger version in an overlay. 

Questions
===
Look at the plugin framework mentioned in #44 as possibly making this feature easier.

**Comment by mikelax** [comment]
@nazar  I haven't done much work on the editor yet, but if you give me some pointers I might be able to work on this, or at least if you implement one I can do the other.

**Comment by mikelax** [comment]
hey @nazar  I was spending a bit of time trying to get the image upload button to have the correct label, I assume you might have tried and ran into the same issues. 

Here is a code snippet that I think *should* be working. it displays correctly, and if you click "around" the icon (ie the gray border) it works. The issue is if you click on the actual icon itself. 

Line 305-308
```jsx
      <Label as="label" htmlFor={id} className="icon button">
        <Icon name="image" />
        <input type="file" id={id} style={{ display: 'none' }} onChange={addImage}/>
      </Label>
```

Looking at the generated HTML and [this codepen](https://codepen.io/apijay/pen/vJzwdK) I can't see why it's not working.... 😢

**Comment by nazar** [comment]
Hey @mikelax 

That won't work as only the Label that is linked to the input will trigger the file upload. Placing the Icon inside the Label steals the onClick from the parent label. We can't use an Icon here and instead have to use a CSS solution to display an icon as I did [here](https://github.com/mikelax/forgingadventures/blob/85-images/client/src/components/shared/components/AddMessageImageButton/index.js#L32-L34) using this [css](https://github.com/mikelax/forgingadventures/blob/85-images/client/src/components/shared/components/AddMessageImageButton/styl.styl#L6-L8).

It's dirty and super frustrating but that is one of the very few styling options available to use for type=file inputs.

**Comment by mikelax** [comment]
@nazar  I took another look at other editors but there still aren't really any "good" options. 
[react-rte](https://github.com/sstur/react-rte) looks promising but is still really early and missing a lot of basic features, ie. images. 

[this medium-like editor](https://github.com/brijeshb42/medium-draft) also looks promising but appears to have stalled in updates. It is built with draft-js anyway. Maybe we can grok how he deals with blockquotes and images, he also support standard embeds (ie. videos).

**Comment by mikelax** [comment]
> That won't work as only the Label that is linked to the input will trigger the file upload. Placing the Icon inside the Label steals the onClick from the parent label. We can't use an Icon here and instead have to use a CSS solution to display an icon as I did here using this css.

thanks, yeah that is an annoying limitation. I was trying to get the Icon to work because currently I am seeing just a square for the button image, is that what you are seeing? 

<img width="347" alt="screen shot 2018-02-25 at 10 39 34 am" src="https://user-images.githubusercontent.com/2080701/36643368-34efc48c-1a18-11e8-826c-895473a17c3c.png">

**Comment by nazar** [comment]
Closed in #102
